### ISSUE TITLE
loss不是很懂

### ISSUE THREAD

**Issue by henbucuoshanghai** [taohan10200/IIM#19]

**Comment by henbucuoshanghai** [comment]
self.binar_map_loss = (torch.abs(binar_map-mask_gt)).mean()

            self.head_map_loss = F.mse_loss(pre_map, mask_gt)
请问为什么二个map在回归的时候 均和mask_gt计算loss？  那这样的话  推理的时候直接任何一个map不就是最后的结果？为什么二者做差呢？不是很懂  多谢

**Comment by taohan10200** [comment]
两个损失针对的是不同的对象，一个是二值化后的输出，一个是二值化前的，二值化后的输出才是最后用来检测连通域进行定位的输出。二者作差即L1损失。

**Comment by henbucuoshanghai** [comment]
第一个pre_map和gt进行l2 loss？&nbsp; &nbsp;gt是黑白的mask图？&nbsp; &nbsp; 后面二个map做差为何和gt进行l1 loss？&nbsp;&nbsp;

**Comment by taohan10200** [comment]
pre_map和gt计算L2 loss, gt 是[0，1]的mask图。 没有出现两个map做差再和gt算L1 loss的， 用来计算L1 loss的是二值化后的输出binar_map。

**Comment by henbucuoshanghai** [comment]
你好 感谢回复  请问figure1的 图 代码有公开吗？请问在哪里呢？感谢

**Comment by henbucuoshanghai** [comment]
密度图方法是怎样生成groundtruth的？  直接人头中心是一个点，只有一个点吗？密度图的gt云图是怎么得到的？
